platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -307,43 +310,41 @@ private static KeyStore getKeyStore(String location, String password) {
   private static void createSolrCore(
       String url, String coreName, String configFileName, HttpClient httpClient)
       throws IOException, SolrServerException {
-    HttpSolrClient client;
-    if (httpClient != null) {
-      client = new HttpSolrClient(url, httpClient);
-    } else {
-      client = new HttpSolrClient(url);
-    }
 
-    HttpResponse ping = client.getHttpClient().execute(new HttpHead(url));
-    if (ping != null && ping.getStatusLine().getStatusCode() == 200) {
-      ConfigurationFileProxy configProxy =
-          new ConfigurationFileProxy(ConfigurationStore.getInstance());
-      configProxy.writeSolrConfiguration(coreName);
-      if (!solrCoreExists(client, coreName)) {
-        LOGGER.debug("Creating Solr core {}", coreName);
+    try (HttpSolrClient client =
+        (httpClient != null ? new HttpSolrClient(url, httpClient) : new HttpSolrClient(url))) {
 
-        String configFile = StringUtils.defaultIfBlank(configFileName, DEFAULT_SOLRCONFIG_XML);
+      HttpResponse ping = client.getHttpClient().execute(new HttpHead(url));
+      if (ping != null && ping.getStatusLine().getStatusCode() == 200) {
+        ConfigurationFileProxy configProxy =
+            new ConfigurationFileProxy(ConfigurationStore.getInstance());
+        configProxy.writeSolrConfiguration(coreName);
+        if (!solrCoreExists(client, coreName)) {
+          LOGGER.debug("Creating Solr core {}", coreName);
 
-        String solrDir;
-        if (System.getProperty("solr.data.dir") != null) {
-          solrDir = System.getProperty("solr.data.dir");
-        } else {
-          solrDir = Paths.get(System.getProperty("karaf.home"), "data", "solr").toString();
-        }
+          String configFile = StringUtils.defaultIfBlank(configFileName, DEFAULT_SOLRCONFIG_XML);
 
-        String instanceDir = Paths.get(solrDir, coreName).toString();
+          String solrDir;
+          if (System.getProperty(SOLR_DATA_DIR) != null) {
+            solrDir = System.getProperty(SOLR_DATA_DIR);
+          } else {
+            solrDir = Paths.get(System.getProperty("karaf.home"), "data", "solr").toString();
+          }
 
```

Comment:
✏️  We might want a debug or trace statement specifying what the solr directory ended up being set to.